include/seqan3/range/decorator/gap_decorator.hpp

```diff
@@ -1038,6 +656,389 @@ template <std::ranges::view urng_t>
 gap_decorator(urng_t range) -> gap_decorator<urng_t>;
 //!\}
 
+/*!\brief A special iterator that enforces std::random_access_iterator albeit its complexity is logarithmic.
+ *
+ * \details
+ *
+ * \attention This iterator models std::random_access_iterator but **does not guarantee constant time access**.
+ *            Random access is provided in \f$O(\log k)\f$ time (where `k` is the number of gaps).
+ *
+ * This iterator returns values when dereferenced, not references, i.e. it does not model the
+ * [Cpp17random_access_iterator](https://en.cppreference.com/w/cpp/named_req/random_access_iterator)
+ * requirements of the STL. It does model the C++20 std::random_access_iterator.
+ *
+ * You can get a view over the gap_decorator with this random access iterator by using
+ * seqan3::views::pseudo_random_access. But note that after adapting the gap_decorator the non-range interfaces
+ * are lost (e.g. insert_gap or erase_gap). This adaptor allows to use the gap_decorator as a range in algorithms
+ * like std::ranges::distance and get the more efficient logarithmic implementation. The regular iterator only
+ * supports std::bidirectional_iterator and thus linear time would be needed to calculate the distance.
```

Comment:
```suggestion
 * is only identified std::bidirectional_iterator and thus linear time would be needed to calculate the distance.
```
since it does support random access but is not identiefied as asuch via the tag..?

Reply:
I adapted your version a little bit